src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,426 @@
+            tabOrder.push(newTab.id);
+        });
+        this.tabOrder.set(newServer.id, tabOrder);
+
+        this.emit(SERVERS_URL_MODIFIED, [newServer.id]);
+        this.persistServers();
+        return newServer;
+    }
+
+    editServer = (serverId: string, server: Team) => {
+        const existingServer = this.servers.get(serverId);
+        if (!existingServer) {
+            return;
+        }
+
+        let urlModified;
+        if (existingServer.url.toString() !== urlUtils.parseURL(server.url)?.toString()) {
+            urlModified = () => this.emit(SERVERS_URL_MODIFIED, [serverId]);
+        }
+        existingServer.name = server.name;
+        existingServer.updateURL(server.url);
+        this.servers.set(serverId, existingServer);
+
+        this.tabOrder.get(serverId)?.forEach((tabId) => {
+            const tab = this.tabs.get(tabId);
+            if (tab) {
+                tab.server = existingServer;
+                this.tabs.set(tabId, tab);
+            }
+        });
+
+        urlModified?.();
+        this.persistServers();
+    }
+
+    removeServer = (serverId: string) => {
+        this.tabOrder.get(serverId)?.forEach((tabId) => this.tabs.delete(tabId));
+        this.tabOrder.delete(serverId);
+        this.lastActiveTab.delete(serverId);
+
+        const index = this.serverOrder.findIndex((id) => id === serverId);
+        this.serverOrder.splice(index, 1);
+        this.remoteInfo.delete(serverId);
+        this.servers.delete(serverId);
+
+        this.persistServers();
+    }
+
+    toggleTab = (tabId: string, isOpen: boolean) => {
+        const tab = this.tabs.get(tabId);
+        if (!tab) {
+            return;
+        }
+        tab.isOpen = isOpen;
+
+        this.persistServers();
+    }
+
+    updateLastActive = (tabId: string) => {
+        const tab = this.tabs.get(tabId);
+        if (!tab) {
+            return;
+        }
+        this.lastActiveTab.set(tab.server.id, tabId);
+
+        this.currentServerId = tab.server.id;
+
+        const serverOrder = this.serverOrder.findIndex((srv) => srv === tab.server.id);
+        this.persistServers(serverOrder);
+    }
+
+    reloadFromConfig = () => {
+        const serverOrder: string[] = [];
+        Config.predefinedTeams.forEach((team) => {
+            const id = this.initServer(team, true);
+            serverOrder.push(id);
+        });
+        if (Config.enableServerManagement) {
+            Config.localTeams.sort((a, b) => a.order - b.order).forEach((team) => {
+                const id = this.initServer(team, false);
+                serverOrder.push(id);
+            });
+        }
+        this.filterOutDuplicateTeams();
+        this.serverOrder = serverOrder;
+        if (Config.lastActiveTeam) {
+            this.currentServerId = this.serverOrder[Config.lastActiveTeam];
+        } else {
+            this.currentServerId = this.serverOrder[0];
+        }
+    }
+
+    private filterOutDuplicateTeams = () => {
```

Comment:
for some reason this function seems to be related to Servers and not `Teams` which is throwing me off a bit